### ISSUE TITLE
Deprecate functionality that is now in imgpkg

### ISSUE THREAD

**Issue by joaopapereira** [carvel-dev/kbld#92]
**Describe the problem/challenge you have**
Some of the functionality that already is part of `imgpkg` should be deprecated from `kbld`

**Describe the solution you'd like**
Scenario 1:

**Given** I have `kbld` installed
**When** I execute `kbld --help`
**Then** I should not see reference to `relocate`, `package` or `unpackage`

Scenario 2:

**Given** I have `kbld` installed
**When** I execute `kbld relocate --help`
**Then** I should see
```
Command "relocate" is deprecated, please use 'imgpkg copy', learn more in https://carvel.dev/imgpkg/docs/latest/commands/#copy
```

Scenario 3:

**Given** I have `kbld` installed
**When** I execute `kbld package --help`
**Then** I should see
```
Command "package" is deprecated, please use 'imgpkg push', learn more in https://carvel.dev/imgpkg/docs/latest/commands/#push
```

Scenario 4:

**Given** I have `kbld` installed
**When** I execute `kbld unpackage --help`
**Then** I should see
```
Command "unpackage" is deprecated, please use 'imgpkg pull', learn more in https://carvel.dev/imgpkg/docs/latest/commands/#pull
```

Scenario 5:

**Given** I access https://carvel.dev/kbld
**When** I click on the `Packaging & Relocation` link
**Then** I should see
```
Packaging and Relocation

This functionality was deprecated in `kbld` and will be removed in the near future. Check the new tool [imgpkg](https://carvel.dev/imgpkg).
```

**Anything else you would like to add:**
In this initial story, we should hide the commands and in the future remove them

**Comment by cppforlife** [comment]
i wonder if it's also worth hiding pkg/unpkg/relocate commands from `kbld -h`? if not then may be also adding `[deprecated]` prefix to command description.

**Comment by danielhelfand** [comment]
I am wondering if it would be worthwhile to write a message to stderr with a deprecation warning. Maybe also mentioning `imgpkg` docs in the message. This would be in addition to hiding the commands from help. 

Do we have any recommended time line for fully removing commands from other Carvel tools?

**Comment by joaopapereira** [comment]
> I am wondering if it would be worthwhile to write a message to stderr with a deprecation warning. Maybe also mentioning imgpkg docs in the message. This would be in addition to hiding the commands from help.

Hmmm, I think we get that for free if we set the cobra command as deprecated, using the deprecated field in the struct. But it is good to ensure that.

> Do we have any recommended time line for fully removing commands from other Carvel tools?

According to our release process document, we say 3 minor releases but we did not have a major discussion to understand if this timing makes sense to our type of tools that are more long living

**Comment by joaopapereira** [comment]
Fixed in both PRs associated

**Comment by joaopapereira** [comment]
This will be released on version v0.30.0
